python/arcticdb/version_store/_normalization.py

```diff
@@ -178,12 +187,19 @@ def _to_primitive(arr, arr_name, dynamic_strings, string_max_len=None, coerce_co
 
     if len(arr) == 0:
         if coerce_column_type is None:
-            raise ArcticNativeNotYetImplemented(
-                "coercing column type is required when empty column of object type, Column type={} for column={}"
-                .format(arr.dtype, arr_name)
-            )
-        else:
-            return arr.astype(coerce_column_type)
+            if IS_PANDAS_TWO:
+                # Before Pandas 2.0, empty series' dtype was float, but as of Pandas 2.0. empty series' dtype became object.
+                # See: https://github.com/pandas-dev/pandas/issues/17261
+                # We want to maintain consistent behaviour, so we treat empty series as containing floats.
+                # val_type = ValueType::FLOAT;
+                coerce_column_type = float
```

Comment:
We have to consider #224 first.

The concern is if the user did not intend for the column to be `float` but happened to have no data for the initial write, then it wouldn't be possible to append to this symbol subsequently.

Reply:
Is there a way to change the type of an (empty) column?

Reply:
I would wait for https://github.com/man-group/ArcticDB/pull/646/ to be merged before this PR first. What do you think?

Reply:
Can you just get rid of coerce_column_type entirely, and all associated tests, and we'll just record a NoneType now that 646 is merged?

Reply:
Done in https://github.com/man-group/ArcticDB/pull/804.